src/datastore/DataStore.ts

```diff
@@ -192,5 +192,9 @@ export interface DataStore {
 
     getRoomCount(): Promise<number>;
 
+    getAccountFirstSeen(userId: string): Promise<Date|null>;
```

Comment:
I've noticed that there's a `data` json blob on users stored in postgres, but i also don't know how to get specifically at that blob of data, and i also don't know which db functions i should be using to specifically look up remote users

Reply:
Ugh the JSON blob is a bit of a sadness. 

For postgres, I'd probabaly crib from what we did [here](https://github.com/matrix-org/matrix-appservice-irc/blob/jess/delay-joins/src/datastore/postgres/schema/v7.ts) and create a new table that stores first seen times (OR: we just make the activity table store a first and latest time, which actually might be more sensible wdyt?)

For nedb: I hate NEDB. I'd crib from `updateLastSeenTimeForUser(userId: string)` and do exactly the same thing.